octobot_trading/producers/simulator/recent_trade_updater_simulator.py

```diff
@@ -46,17 +54,45 @@ def __init__(self, channel, importer):
                     self.last_timestamp_pushed = recent_trades_data[0]
                     await self.push(pair, recent_trades_data[-1])
         except DataBaseNotExists as e:
-            self.logger.warning(f"Not enough data : {e}")
+            self.logger.warning(f"Not enough data : {e} will use ohlcv data to simulate recent trades.")
             await self.pause()
             await self.stop()
-        except IndexError as e:
-            self.logger.warning(f"Failed to access recent_trades_data : {e}")
+
+    async def _register_on_recent_trades(self):
+        ohlcv_chan = get_exchange_chan(OHLCV_CHANNEL, self.channel.exchange_manager.id)
+        for producer in ohlcv_chan.get_producers():
```

Comment:
Why do you need to wait that ohlcv producer to be ready ?

Reply:
Yes, otherwise it will get called by historical candles push during initialization, which can lead in invalid values

Reply:
Oh yes ok I see, we have to have the candles related to the current backtesting time to generate recent trades :+1:

Reply:
Can you add some comments in the code to explain this please ? Thanks